eras/shelley/test-suite/test/Test/Cardano/Ledger/Shelley/Examples/Combinators.hs

```diff
@@ -175,30 +176,74 @@ newLab b cs =
 
 -- | = Update Fees and Deposits
 --
--- Update the fee pot and deposit pot with the new fees
--- and the change to the deposit pot.
--- Note: do not use this function when crossing the epoch boundary,
+-- Update the fee pot and deposit pot with the new fees and deposits
+-- adjust the deposit tables in the DPState.
+-- Notes
+--   1) do not give this function duplicates in the 'stakes' or 'pools' inputs.
+--   2) do not use this function when crossing the epoch boundary,
 -- instead use 'newEpoch'.
 feesAndDeposits ::
   forall era.
+  ShelleyPParams era ->
   Coin ->
+  [Credential 'Staking (EraCrypto era)] ->
+  [PoolParams (EraCrypto era)] ->
+  ChainState era ->
+  ChainState era
+feesAndDeposits ppEx newFees stakes pools cs = cs {chainNes = nes'}
+  where
+    nes = chainNes cs
+    es = nesEs nes
+    ls = esLState es
+    DPState dstate pstate = lsDPState ls
+    utxoSt = lsUTxOState ls
+    utxoSt' =
+      utxoSt
+        { utxosDeposited =
+            (utxosDeposited utxoSt)
+              <+> (length stakes <×> _keyDeposit ppEx)
+              <+> (newcount <×> _poolDeposit ppEx),
+          utxosFees = (utxosFees utxoSt) <+> newFees
+        }
+    ls' = ls {lsUTxOState = utxoSt', lsDPState = dpstate'}
+    -- Count the number of new pools, because we don't take a deposit for existing pools
+    -- This strategy DOES NOT WORK if there are duplicate PoolParams in one call
+    newcount = foldl accum 0 pools
+    accum n x = if Map.member (ppId x) (psDeposits pstate) then (n :: Integer) else n + 1
```

Comment:
```suggestion
    newcount = length $ filter (\x -> ppId x `Map.notMember` psDeposits pstate) pools
```

Reply:
I like it the way it is. It suggests that we are accumulating individual pool deposits, which is exactly what we are doing.

Reply:
Then you should switch `foldl` to `foldl'` to avoid memory leaks

Reply:
Also, you aren't really accumulating deposits, you are just counting here ppIds that aren't in the deposits pstate

Reply:
I changed it to foldl'